### ISSUE TITLE
autoFit text option has no effect?

### ISSUE THREAD

**Issue by DSheffield** [gitbrent/PptxGenJS#312]
I'm trying to create a text box with the autofit option, but it does not appear to be working.  The box always has a fixed size and the text overflows outside of it.  Am I missing something?

```
const PptxGenJS = require("pptxgenjs");

let pptx = new PptxGenJS();
let slide = pptx.addNewSlide();

slide.addText('The quick brown fox jumps over the lazy dog',
     { isTextBox: true, autoFit: true, x: 1.0, y: 1.0, w: 1.0, fontSize: 12});

pptx.save('SamplePresentation');
```

**Comment by 053steve** [comment]
Hi, I'm also getting that problem. Have u found out what was wrong?

**Comment by DSheffield** [comment]
My suspicion at this point is that PPT does not apply the autofit transformation on opening the presentation, but rather only when you edit the text in the shape, because:

1.  If you generate a shape via the library and then create an identical slide in PPT, the resulting XML is almost identical and doesn't seem to have any semantic differences.

2.  If you generate a shape via the library, open the .pptx file, and edit the text in the shape in any way, then PPT does resize the shape at that point to match the resulting text.

**Comment by gitbrent** [comment]
@DSheffield,

You are correct.  Only the PowerPoint runtime can read the text/formatting of a shape's contents and fit the shape's `w` and `h` to that calculated size.  There's no code i can add that'll accomplish the same outcome.  OpenXML only goes so far.

Lastly, the `autofit` option does work for sizing the Textbox Shape's width given the correct options, so it's not a matter of it not working, it's just a limited option.
